CMakeLists.txt

```diff
@@ -57,10 +57,29 @@ list(APPEND CMAKE_MODULE_PATH
     $ENV{ROCM_PATH}/llvm
 )
 
+# Detect if already using HIP language
+get_property(ENABLED_LANGUAGES GLOBAL PROPERTY ENABLED_LANGUAGES)
+if(HIP IN_LIST ENABLED_LANGUAGES)
```

Comment:
Could we maybe just finish up #266 and suggest that users to set `-DROCRAND_CMAKE_HIP_LANGUAGE=ON`?